### ISSUE TITLE
Add top-level class to the IDE's parent container

### ISSUE THREAD

**Issue by gadenbuie** [rstudio/rstudio#10216]
The recent changes to the theming in #10081 (which are generally great and I'm happy to see!) removed the `.rstudio-themes-flat` class from the `<body>` tag. 

On Mac with a light theme, for example, I see `<body class="macintosh avoid-move-cursor rstudio-themes-light-menus">`. All of these classes appear to be communicate current system or state, so any CSS rules that use them will have to take this into account -- at minimum at least two rules would be required for light and dark, choosing from `.rstudio-themes-{light,dark}-menus`.

Even if the IDE is always contained in a `<body>` tag, it'd be helpful for theming purposes to have a stable class on this element. This would make long-term maintenance a little easier since I won't need to Find and Replace `.rstudio-themes-light-menus` if the purpose of that class changes, or have two selectors for the same purpose. Having a class, rather than the `body` element, in the selector helps with specificity issues and to avoid having to use `!important` (which tends to create more problems unless the rule is specific enough where a class selector is better). Previously, the `.rstudio-themes-flat` class served this purpose for me.

**Comment by gadenbuie** [comment]
Actually, `.rstudio-themes-dark-menus` alone won't work because the class can dynamically change to `.rstudio-themes-dark-menus-disabled` depending on IDE state, e.g. when Git > Pull adds an overlay menu.

**Comment by jmcphers** [comment]
Yes, I removed this because `rstudio-themes-flat` is now totally redundant (all themes are flat). 

Do you need a selector on `<body>` _specifically_? Or just something under which all the UI is guaranteed to be in scope? (was wondering if something like `rstudio-container` on an outer `<div>` would be cleaner)

**Comment by gadenbuie** [comment]
Either one would work! Having an `.rstudio-container` would be perfect, I don't think the specific element matters as much as having the class on the outermost container to scope the UI root element... as long as the `div.rstudio-container` would also hold pop-up windows like the Global Options editor. It'd also be nice if the state-based classes were added to the container div, too, allowing something like `.rstudio-container.macintosh`.

**Comment by gadenbuie** [comment]
I just wanted to check in that we're really intending to put this off until 2023. My ask is pretty basic: add a stable class to the IDE's parent element. (As to the question of _which_ element, I think whichever element previously recieved `.rstudio-themes-flat` or currently receives `.rstudio-themes-dark-menus` is probably the appropriate element.)

Previously the IDE used the `.rstudio-themes-flat` class on the parent container, which was stable and used by many custom themes as an anchor class and for specificity reasons. Removing that class is going to break many themes without offering a clear alternative, since the classes on parent container are now all dynamic.

A quick GitHub search shows more than 1,000 uses of the `.rstudio-themes-flat` class: https://github.com/search?q=.rstudio-themes-flat&type=code
